src/Bicep.Core/TypeSystem/DeclaredTypeManager.cs

```diff
@@ -228,8 +228,8 @@ private DeclaredTypeAssignment GetParameterType(ParameterDeclarationSyntax synta
             if (!binder.FileSymbol.TryGetBicepFileSemanticModelViaUsing().IsSuccess(out var semanticModel, out var failureDiagnostic))
             {
                 // failed to resolve using
-                return failureDiagnostic is ErrorDiagnostic error
-                    ? ErrorType.Create(error)
+                return failureDiagnostic.Level == DiagnosticLevel.Error
```

Comment:
Might be worth it to have an IsErrorDiagnostic() or IsErrorLevel() extension for IDiagnostic.

Reply:
Good suggestion! I have a follow-up PR - will address this there.